File: SWEA/D3/1289. 원재의 메모리 복구하기/원재의 메모리 복구하기.py
t = int(input())
for tc in range(1, t+1):
    memory = input()
    flag = '0'
    answer = 0

    for i in range(len(memory)):
        if memory[i] != flag:
            answer += 1
            flag = memory[i]

    print(f'#{tc} {answer}')

File: SWEA/D3/1289. 원재의 메모리 복구하기/원재의 메모리 복구하기.java
import java.util.Scanner;
public class Solution {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int t = sc.nextInt();
		
		for(int tc = 1; tc <= t; tc++) {
			String memory = sc.next();
			int answer = 0;
			char num = '0';
			
			for (int i = 0; i < memory.length(); i++) {
				if(memory.charAt(i) != num) {
					answer++;
					num = memory.charAt(i);
				}
			}
			System.out.printf("#%d %d",tc, answer);
            System.out.println();
		}
	}

}